runtime/interface.go

```diff
@@ -41,93 +43,110 @@ type Block struct {
 type ResolvedLocation = sema.ResolvedLocation
 type Identifier = ast.Identifier
 type Location = common.Location
+type AddressLocation = common.AddressLocation
 
-type Interface interface {
-	// ResolveLocation resolves an import location.
-	ResolveLocation(identifiers []Identifier, location Location) ([]ResolvedLocation, error)
-	// GetCode returns the code at a given location
-	GetCode(location Location) ([]byte, error)
-	// GetCachedProgram attempts to get a parsed program from a cache.
-	GetCachedProgram(Location) (*ast.Program, error)
-	// CacheProgram adds a parsed program to a cache.
-	CacheProgram(Location, *ast.Program) error
-	// GetValue gets a value for the given key in the storage, owned by the given account.
-	GetValue(owner, key []byte) (value []byte, err error)
-	// SetValue sets a value for the given key in the storage, owned by the given account.
-	SetValue(owner, key, value []byte) (err error)
-	// CreateAccount creates a new account.
-	CreateAccount(payer Address) (address Address, err error)
-	// AddAccountKey appends a key to an account.
-	AddAccountKey(address Address, publicKey []byte) error
-	// RemoveAccountKey removes a key from an account by index.
-	RemoveAccountKey(address Address, index int) (publicKey []byte, err error)
+type Accounts interface {
+
+	// creates a new account address and set the exists flag for this account
+	CreateAccount(caller Location) (address Address, err error)
+	// Exists returns true if the account exists
+	Exists(address Address) (exists bool, err error)
+
+	// SuspendAccount suspends an account (set suspend flag to true)
+	SuspendAccount(address Address, caller Location) error
+	// UnsuspendAccount unsuspend an account (set suspend flag to false)
+	UnsuspendAccount(address Address, caller Location) error
+	// returns true if account is suspended
+	IsSuspended(address Address) (isSuspended bool, err error)
+
+	// TODO ramtin merge this with 	Code(location Location) error
+	// AccountContractCode returns the code associated with an account contract.
+	ContractCode(address AddressLocation) (code []byte, err error)
 	// UpdateAccountContractCode updates the code associated with an account contract.
-	UpdateAccountContractCode(address Address, name string, code []byte) (err error)
-	// GetAccountContractCode returns the code associated with an account contract.
-	GetAccountContractCode(address Address, name string) (code []byte, err error)
-	// RemoveAccountContractCode removes the code associated with an account contract.
-	RemoveAccountContractCode(address Address, name string) (err error)
-	// GetSigningAccounts returns the signing accounts.
-	GetSigningAccounts() ([]Address, error)
```

Comment:
What happened to this one?

Reply:
it has been moved to Runnable (Authorizers() []Address) should provide the same